src/backend/executor/execIndexing.c

```diff
@@ -243,7 +243,7 @@ ExecCloseIndices(ResultRelInfo *resultRelInfo)
 			continue;			/* shouldn't happen? */
 
 		/* Drop lock acquired by ExecOpenIndices */
-		index_close(indexDescs[i], RowExclusiveLock);
+		index_close(indexDescs[i], get_ENR_withoid(currentQueryEnv, indexDescs[i]->rd_id, ENR_TSQL_TEMP) ? AccessShareLock : RowExclusiveLock);
```

Comment:
can we override the lock in index_close function itself? that way it is a separate line with less chance of merge conflict

Reply:
scratch that - my previous answer doesn't apply at all. I'll move the check inside of index_close()